components/script/dom/servoparser/mod.rs

```diff
@@ -946,3 +956,61 @@ impl TreeSink for Sink {
         vtable_for(&node).pop();
     }
 }
+
+/// https://html.spec.whatwg.org/multipage/#create-an-element-for-the-token
+fn create_element_for_token(
+    name: QualName,
+    attrs: Vec<ElementAttribute>,
+    document: &Document,
+    creator: ElementCreator
+) -> DomRoot<Element> {
+    // Step 3.
+    let is = attrs.iter()
+        .find(|attr| attr.name.local.eq_str_ignore_ascii_case("is"))
+        .map(|attr| LocalName::from(&*attr.value));
+
+    // Step 4.
+    let definition = document.lookup_custom_element_definition(&name.ns, &name.local, is.as_ref());
+
+    // Step 5.
+    // TODO: Check if the parser was created for the HTML fragment parsing algorithm
+    let will_execute_scrtipt = definition.is_some();
+
+    // Step 6.
+    if will_execute_scrtipt {
+        // Step 6.1.
+        // TODO: handle throw-on-dynamic-markup-insertion counter.
+        // Step 6.2
+        // TODO: If the JavaScript execution context stack is empty, then perform a microtask checkpoint.
```

Comment:
We should add an API to https://github.com/servo/servo/blob/master/components/script/dom/bindings/settings_stack.rs to check this.